pkg/controllers/inventory/inventory_controller.go

```diff
@@ -0,0 +1,175 @@
+package inventory
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	commonservice "github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/inventory"
+)
+
+const (
+
+	// How long to wait before retrying
+	minRetryDelay = 5 * time.Second
+	maxRetryDelay = 300 * time.Second
+
+	inventoryGCJitterFactor = 0.1
+)
+
+type WatchResourceFunc func(c *InventoryController, mgr ctrl.Manager) error
+
+var (
+	log                = &logger.Log
+	keyFunc            = cache.DeletionHandlingMetaNamespaceKeyFunc
```

Comment:
could you comment why use this DeletionHandlingMetaNamespaceKeyFunc?

Reply:
done